Algorithms/Sorters/MSDRadixSorter.cs

```diff
@@ -0,0 +1,62 @@
+namespace Algorithms.Sorters
+{
+    /// <summary>
+    ///     Radix sort is a non-comparative sorting algorithm. It avoids comparison by creating
+    ///     and distributing elements into buckets according to their radix.
+    ///     Radix sorts can be implemented to start at either the most significant digit (MSD)
+    ///     or least significant digit (LSD).
+    ///     MSD radix sorts are most suitable for sorting array of strings with variable length
+    ///     in lexicographical order.
+    /// </summary>
+    public class MSDRadixSorter
```

Comment:
Also please rename it to MsdRadixSorter. Acronym rules are discussed here https://softwareengineering.stackexchange.com/questions/186277/should-the-variable-be-named-id-or-id
And same in tests